### DISCUSSION TITLE
I try using STM32 to create bootloader?

### DISCUSSION THREAD

**Discussion by ruyi-tsai** [microsoft/jacdac#870] [General]
I get two elf file? 
  I need  burn bl-servo.elf on STM? 
  what purpose app-servo.elf ? is for servo modules?
1876	     12	   1664	  13552	   34f0	built/jm-v2.1/app-motor.elf
  12008	     12	   1664	  13684	   3574	built/jm-v2.1/app-motor_led.elf
  12196	     12	   1664	  13872	   3630	built/jm-v2.1/app-servo.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-motor.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-motor_led.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-servo.elf

**Comment by jamesadevine** [comment]
Hi @ruyi-tsai,

`bl-<app>.elf` is the bootloader for the specific board you are building for. `app-<service>.elf` is the application binary for the specific board you have built. Combined, they form the complete firmware.

`make full-flash` -- flashes bootloader and app
`make flash` -- flashes only the app
`make flash BL=1` -- flashes only the bootloader

**Comment by ruyi-tsai** [comment]
I try make full-flash -- flashes bootloader and app 
 11876	     12	   1664	  13552	   34f0	built/jm-v2.1/app-motor.elf
  12008	     12	   1664	  13684	   3574	built/jm-v2.1/app-motor_led.elf
  12196	     12	   1664	  13872	   3630	built/jm-v2.1/app-servo.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-motor.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-motor_led.elf
   2904	      8	   1592	   4504	   1198	built/jm-v2.1/bl-servo.elf

ls: cannot access '/dev/cu.usbmodem????????1': No such file or directory
echo "file built/jm-v2.1/bl-servo.elf" > built/debug.gdb

If I want to try use my own module to connect  than I need only burn bootloader?

**Reply by jamesadevine** [reply]
You'll need both the bootloader and the app. 

What debugger are you using? The build system assumes that you are using a variant of [blackmagic probe](https://github.com/blacksphere/blackmagic). An STM32f1 bluepill is compatible with blackmagic probe firmware: https://github.com/mmoskal/blackmagic-bluepill

You can configure the build for other debuggers by playing around with [Makefile.user](https://github.com/microsoft/jacdac-msr-modules/blob/main/sample-Makefile.user#L8).

**Comment by pelikhan** [comment]
@ruyi-tsai are you unblocked? We just release the latest firmware and schematics for jacdac. Do you need help to get unblocked?
